### PR TITLE
Add UseDyeingTube specialized function (packet: 0x95) + Crash on Tracking arrow bugfix

### PR DESCRIPTION
**Pull request by cesare-montresor**
- Added:
  Misc.UseDyeingTub(dyes, dyeingTub, color) to be able to change color for Dyeing Tub programmatically.
  Leverages and extends part of the work already done for HueEntry class ( Options -> Override spell hues -> set -> set in game )

- Fix crash related to Tracking arrow:
  I probably messed it up when I was cleaning things up for autodoc

### CODE REVIEW

File: Razor/Network/Handlers.cs [hunk 1]
```diff
@@ -3025,12 +3026,16 @@ private static void HueResponse(PacketReader p, PacketHandlerEventArgs args)
 			ushort iid = p.ReadUInt16();
 			ushort hue = p.ReadUInt16();
 
-			if (serial == Serial.MinusOne)
-			{
-				if (HueEntry.Callback != null)
-					HueEntry.Callback(hue);
-				args.Block = true;
-			}
+            if (serial == Serial.MinusOne || HueEntry.Callback != null)
```
Comment (credzba) [inline Razor/Network/Handlers.cs:3029]: Dalamar this logic and the old logic are not equivalent.  Did you mean to change the logic flow ?

File: Razor/RazorEnhanced/Misc.cs [hunk 2]
```diff
@@ -967,6 +967,26 @@ public static HotKeyEvent LastHotKey()
...
             return HotKeyEvent.LastEvent;
         }
+        
+        /// <summary>
+        /// Use the Dye Tube via packets. Need to specify the dyes, the dye tube and the color to use.
+        /// </summary>
+        /// <param name="dyes">Dyes as Item object.</param>
+        /// <param name="dyeTube">DyeTube as Item object.</param>
+        /// <param name="color">Color to choose.</param>
+        public void UseDyeTube(Item dyes, Item dyeTube, int color) {
```
Comment (credzba) [inline Razor/RazorEnhanced/Misc.cs]: DyeTub - there is no e

File: Razor/RazorEnhanced/Misc.cs [hunk 3]
```diff
@@ -967,6 +967,26 @@ public static HotKeyEvent LastHotKey()
         {
             return HotKeyEvent.LastEvent;
         }
+
```
Comment (credzba) [inline Razor/RazorEnhanced/Misc.cs]: I agree putting this in Items would be better

**Review by credzba — CHANGES_REQUESTED** [review]

File: Razor/Network/Handlers.cs (hunk 1 above)
Comment (cesare-montresor) [reply Razor/Network/Handlers.cs:3029]: Yes, the pre-existing case were using Serial.MinusOne to identify an "artificial packet" sent by RE to be able to use the in-game color picker for selecting the hue ( that's why it's blocking args.Block) and that part is still achieved.

There is only one of the pre-existing cases where HueEntry.Callback is set to something other then null, which is the same case of Serial.MinusOne.
Besides, the logic should still hold in more general cases: "if there is a Callback set on change color, don't prompt the client" as anyway the color picker is not dismissable ( you will be forced to pick another color )

File: Razor/RazorEnhanced/Item.cs [hunk 4]
```diff
@@ -543,12 +544,27 @@ public void ChangeDyeingTubColor(Item dyes, Item dyeingTub, int color)
...
+                    (serial, iid, hue) =>
+                    {
+                        HueEntry.Callback = null;
+                        Assistant.Client.Instance.SendToServer(new HuePicker(serial, iid, (ushort)color));
+                    }
+                );
+                HueEntry.Callback = cb;
                 Target.TargetExecute(dyeingTub);
+
+                // Dalamar: Callback AutoCleanup
```
Comment (cesare-montresor) [inline Razor/RazorEnhanced/Item.cs:557]: Added autocleanup feature

File: Razor/RazorEnhanced/Misc.cs (hunk 3 above)
Comment (cesare-montresor) [reply Razor/RazorEnhanced/Misc.cs]: Moved to Items.ChangeDyeingTubColor() for clarity

File: Razor/RazorEnhanced/Misc.cs (hunk 2 above)
Comment (cesare-montresor) [reply Razor/RazorEnhanced/Misc.cs]: fixed spelling ^_^

**Review by credzba — APPROVED** [review]
Looks great

**Pull request merged**